src/core/xfa/formcalc_lexer.js

```diff
@@ -239,8 +239,8 @@ class Lexer {
 
   getNumber(first) {
     const match = this.data.substring(this.pos).match(numberPattern);
-    if (!match) {
-      return first - 0x30 /* = 0 */;
+    if (!match[0]) {
```

Comment:
[According to MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#return_value) `String.prototype.match` can return `null` as well, don't we need to handle that case in this method?